### ISSUE TITLE
Rename "criterion" to "loss function" throughout code/docs

### ISSUE THREAD

**Issue by rwood-97** [Living-with-machines/MapReader#468]
**Is your feature request related to a problem? Please describe.**
I am more used to hearing "loss function" used to describe the loss function/criterion used when training a model. Within our code we call this "criterion" which technically is correct but is less commonly used. 

To avoid confusion it would be good to change the use of "criterion" to "loss_function" throughout the code and docs.

@kallewesterling @kmcdono2 do you agree with this?

**Comment by kmcdono2** [comment]
Yes, let's make that change.

**Comment by kallewesterling** [comment]
Yep, in another package that I was working on, we used `loss` and `loss_fn`, see:
- https://alan-turing-institute.github.io/deepsensor/reference/model/model.html#deepsensor.model.model.ProbabilisticModel.loss
- https://alan-turing-institute.github.io/deepsensor/reference/model/convnp.html#deepsensor.model.convnp.ConvNP.loss_fn
